assignments/array-methods.js

```diff
@@ -54,30 +54,53 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
+ 
+let fullNames = [];
+runners.forEach(function (racer, index){
+  fullNames.push(racer.first_name + " " + racer.last_name)
+})
+console.log(fullNames);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps = runners.map(function(racer){
+  return racer.first_name.toUpperCase();
+});
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(function(racer){
+  return racer.shirt_size === "L"
+});
 console.log(largeShirts);
 
+
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((acc, runner) => acc + runner.donation, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 persons name and donation
+let nameDonation = [];
+runners.forEach(function (racer){
+  nameDonation.push(racer.first_name + " $" + racer.donation);
+});
+console.log(nameDonation);
 
-// Problem 2
+// Problem 2 get emails to send out a reminder
+let emailList = runners.map(function(racer){
+  return racer.email;
+});
+console.log(emailList); 
 
-// Problem 3
\ No newline at end of file
+// Problem 3 name and company
+let nameCo = [];
+runners.forEach(function (racer){
+  nameCo.push(racer.first_name + " " + racer.company_name);
```

Comment:
This could also be done with template literals and string interpolation.

`${racer.first_name } ${racer.company_name}` etc.